### PR TITLE
fix(cspell): added cspell configuration

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction ✏️ 
Improve code commit quality

## Resolution ✔️ 
* Introduced `cspell`
* Introduced `lint-staged` git hook commands ranging from linting to spell check.
* Removed all `MULESOFT_` reference and replaced it with `APIM_`

## Miscellaneous ➕ 
* Mitigated few GitHub alerts dependencies issues.
* Update documentation.
* Added Cypress badge to the main `README.md`

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        "huskyrc",
+        "Informatica",
+        "INITDB",
+        "Initialised",
+        "isready",
+        "jspf",
+        "limitter",
+        "lintstagedrc",
+        "localise",
+        "Markan",
```
Comment (ttbarnes) [inline cspell.json:70]: Do we need this in here?

File: cspell.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        "typeorm",
+        "Typeorm",
+        "typescript",
+        "ukef",
+        "UKEF",
+        "uksouth",
+        "Unauthorised",
+        "venv",
+        "VNET",
+        "XLXS"
```
Comment (ttbarnes) [inline cspell.json:101]: `XLSX`?

File: src/api/package.json [hunk 3]
```diff
@@ -33,38 +36,41 @@
     "@graphql-tools/schema": "^10.0.0",
     "@keystone-6/auth": "^7.0.2",
     "@keystone-6/core": "^5.5.1",
-    "@keystone-6/fields-document": "^8.0.0",
+    "@keystone-6/fields-document": "^8.0.1",
     "@types/express": "^4.17.17",
-    "axios": "^1.4.0",
-    "axios-mock-adapter": "^1.21.5",
```
Comment (ttbarnes) [inline src/api/package.json:39]: We need this, used for some API tests

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -162,15 +162,20 @@ jobs:
...
     environment:
       name: ${{ needs.setup.outputs.environment }}
     runs-on: ubuntu-latest
 
     strategy:
       fail-fast: true
+      # Single dimension matrix - 4 jobs
       matrix:
-        machines: [1, 2, 3, 4, 5, 6, 7, 8]
+        machines: [1, 2, 3, 4]
```
Comment (ttbarnes) [inline .github/workflows/test.yml:174]: Does halving the machines increase the time it takes to run the tests? I think we currently have just under 300 insurance E2E tests.

File: README.md [hunk 5]
```diff
@@ -1,26 +1,33 @@
-# Export Insurance Policies
+# Export Insurance Policies (EXIP) :briefcase:
+Welcome to the Export Insurance Policies repository! This repository houses the code for the UK Export Finance (UKEF) EXIP service, a public-facing application that offers various features to exporters:
 
-This repository contains the code for the UK Export Finance EXIP service - a public facing application that allows exporters to:
+- Obtain a quote for export insurance coverage with UKEF.
```
Comment (ttbarnes) [inline README.md:4]: Should we be aligned with the actual service name here (before this change it's "Get a quote", same as the service name)

File: README.md [hunk 6]
```diff
@@ -149,195 +129,86 @@ export const lists = {
...
+Middleware fetches the application from the API when a page is loaded. This middleware uses Apollo to make a GraphQL query to the API. The API automatically handles the request and returns the application data, which is then assigned to `res.locals` for use by the subsequent controller.
 
-### When an application page is loaded
+### When a User Submits a Form in the Application Flow :computer:
 
-We have some middleware that fetches the application from the API. This middleware calls the API via [this file](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/middleware/insurance/get-application/index.ts#L63). We use apollo to run the GraphQL query.
+When a user submits
 
-The API automatically handles the request (thanks to Keystone) and returns the application.
+ a form, the following process occurs:
```
Comment (ttbarnes) [inline README.md:168]: One line instead of 166 and 168?

File: README.md [hunk 7]
```diff
@@ -149,195 +129,86 @@ export const lists = {
...
-
-- [global content strings](https://github.com/UK-Export-Finance/exip/tree/main-application/src/ui/server/content-strings) (buttons, links etc)
-- [page content strings](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/content-strings/pages/index.ts)
-- [field content strings](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/content-strings/fields/index.ts)
-
-Note: field IDs are in [constants](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/constants) because an ID is not a content string.
-
-## DRY UI controllers and rendering
-
-Every page/controller requires the same content and links in order to meet design. For example, every page has a header, footer, service name and a back link. To save repetition, we've introduced **page variables functions** that should be used in every single controller.
```
Comment (ttbarnes) [inline README.md:275]: Are the examples of page variables etc intentionally removed? A lot is happening with these and has been confusing for new developers, which is why it was added to the README

Comment (abhi-markan) [reply README.md:275]: Apologies this has been removed by mistake, have added it back.

File: README.md [hunk 8]
```diff
@@ -365,75 +236,62 @@ export const get = (req: Request, res: Response) =>
...
-All newly created application's automatically use the latest version number.
+This approach simplifies the handling of different versions of the application and avoids complex logic in the codebase or other internal systems.
 
-An example of possible versions:
+These are the key aspects of the UK Export Finance EXIP service codebase and development process. If you have specific questions or need more details about any particular aspect, feel free to ask!
 
-- Version number 1: MVP, no support for applications over £100.
-- Version number 2: Support for applications over £100.
-- Version number 3: Payments integration
+---
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply README.md:374]: Nope this is to end the document in MD syntax.

File: src/api/README.md [hunk 9]
```diff
@@ -1,31 +1,41 @@
...
 
-## Files that should not be changed
+These files are essential for the functioning of your Keystone-based API, and any changes should be made through appropriate Keystone configurations and not by editing these files directly.
 
-The following files are automatically generated by Keystone and should not be changed manually:
+If you have any specific questions or need further guidance related to this API or KeystoneJS, please feel free to ask.
 
-- `/schema.graphql`
-- `/schema.prisma`
+---
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply src/api/README.md:41]: MD syntax line.

File: src/ui/README.md [hunk 10]
```diff
@@ -1,3 +1,32 @@
...
+  - **api**: This directory contains the application's APIs.
+    - **external**: This directory contains APIs that are provided by external services.
+    - **keystone**: This directory contains APIs that are provided by the Keystone framework.
+  - **constants**: This directory contains the application's constants.
+  - **content-strings**: This directory contains the application's content strings.
+  - **controllers**: This directory contains the application's controllers.
+  - **middleware**: This directory contains the application's middleware.
+  - **models**: This directory contains the application's models.
+  - **services**: This directory contains the application's services.
+
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply src/ui/README.md:24]: Have added.

File: .github/workflows/test.yml (hunk 4 above)
Comment (abhi-markan) [reply .github/workflows/test.yml:174]: Testing various combinations.

**Comment by abhi-markan** [comment]
Closing this PR due to various failing errors, this PR will be abstracted into new PRs.

**Pull request closed without merging**